FinSearchUnified/tests/BusinessLogic/ExportTest.php

```diff
@@ -0,0 +1,347 @@
+            'Other shop articles are not considered' => [
+                'Active' => [true, true],
+                'Shopkey' => 'ABCD0815',
+                'start' => 0,
+                'count' => null,
+                'hideNoInStock' => true,
+                'laststock' => 1,
+                'instock' => 5,
+                'minpurchase' => 3,
+                'categories' => [39, true],
+                'expected' => 0,
+                'Expected that variation are not returned'
+            ],
+            'Only article with active category is considered' => [
+                'Active' => [true, true],
+                'Shopkey' => 'ABCD0815',
+                'start' => 0,
+                'count' => null,
+                'hideNoInStock' => true,
+                'laststock' => 1,
+                'instock' => 5,
+                'minpurchase' => 3,
+                'categories' => [$shopCategoryId, false],
+                'expected' => 0,
+                'Expected that variation is not returned'
+            ]
+        ];
+    }
+
+    /**
+     * Method to run the export test cases using the data provider
+     *
+     * @dataProvider articleProvider
+     *
+     * @param array $isActive
+     * @param string $shopkey
+     * @param int $start
+     * @param int $count
+     * @param bool $hideNoInStock
+     * @param int $laststock
+     * @param int $instock
+     * @param int $minpurchase
+     * @param array $category
+     * @param int|null $expected
+     * @param string $errorMessage
+     *
+     * @throws \Exception
+     */
+    public function testArticleExport(
+        $isActive,
+        $shopkey,
+        $start,
+        $count,
+        $hideNoInStock,
+        $laststock,
+        $instock,
+        $minpurchase,
+        $category,
+        $expected,
+        $errorMessage
+    ) {
+        $categoryRepository = Shopware()->Models()->getRepository('Shopware\Models\Category\Category');
+
+        /** @var Category $categoryModel */
+        $categoryModel = $categoryRepository->find($category[0]);
+        $this->assertInstanceOf(
+            'Shopware\Models\Category\Category',
+            $categoryModel,
+            sprintf('Could not find category for given ID %d', $category[0])
+        );
+
+        $categoryModel->setActive($category[1]);
+        Shopware()->Models()->persist($categoryModel);
+        Shopware()->Models()->flush();
+
+        // Create articles with the provided data to test the export functionality
+        for ($i = 0; $i < count($isActive); $i++) {
+            $this->createTestProduct($i, $isActive[$i], $laststock, $instock, $minpurchase, $categoryModel->getId());
+        }
+
+        Shopware()->Config()->offsetSet('hideNoInStock', $hideNoInStock);
+        Shopware()->Config()->offsetSet('ShopKey', 'ABCD0815');
+
+        /** @var Export $exportService */
+        $exportService = new Export(Shopware()->Container()->get('shopware_searchdbal.dbal_query_builder_factory'));
```

Comment:
Since this is the same for every test case:
1. Make `$exportService` a private member.
2. Initialize it in the `setUp` method.